tests/unit/backend/layers/api/test_curation_api.py

```diff
@@ -1202,31 +1200,27 @@ def test__new_from_link__OK(self, *mocks):
         Calling PUT /datasets/:dataset_id should succeed if a valid link is uploaded by the owner of the collection
         """
 
-        with self.subTest("with version_ids"):
-            dataset = self.generate_dataset(
-                statuses=[DatasetStatusUpdate(DatasetStatusKey.PROCESSING, DatasetProcessingStatus.INITIALIZED)],
-            )
+        def _test_create(collection_id, dataset_id):
             body = {"link": self.good_link}
             headers = self.make_owner_header()
             response = self.app.put(
-                f"/curation/v1/collections/{dataset.collection_version_id}/datasets/{dataset.dataset_version_id}",
+                f"/curation/v1/collections/{collection_id}/datasets/{dataset_id}",
                 json=body,
                 headers=headers,
             )
             self.assertEqual(202, response.status_code)
 
-        with self.subTest("with collection_ids"):
+        with self.subTest("with version_ids"):
             dataset = self.generate_dataset(
```

Comment:
nit: Do you need to generate a new dataset for each test?

Reply:
I think it's better if each test is completely independent, including subtests.